packages/fxa-event-broker/lib/db/in-memory.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Datastore } from './index';
+
+class InMemoryDatastore implements Datastore {
+  private users: { [key: string]: { [key: string]: boolean } };
+
+  constructor(config: {}) {
+    this.users = {};
+  }
+
+  public async storeLogin(uid: string, clientId: string) {
```

Comment:
The annotation isn't needed in TypeScript since the interface defines it. All the tooling works as expected as well. The Typescript interface doesn't support specifying it as `async` which is why the Promise return type is used, while the implementation may use async still. Most of the examples I've found don't bother with annotating implementations since the interface and typechecker validate it already.